src/Units/UnitModel.java

```diff
@@ -6,88 +6,115 @@
 import java.util.Set;
 import java.util.Vector;
 
-import Engine.GameAction;
 import Engine.UnitActionType;
+import Engine.GameEvents.GameEventQueue;
 import Terrain.Location;
+import Terrain.MapMaster;
 import Terrain.TerrainType;
 import Units.MoveTypes.MoveType;
-import Units.Weapons.WeaponModel;
 
 /**
  * Defines the invariant characteristics of a unit. One UnitModel can be shared across many instances of that Unit type.
  */
-public class UnitModel implements Serializable
+public abstract class UnitModel implements Serializable
 {
   private static final long serialVersionUID = 1L;
 
-  public enum UnitEnum
+  /**
+   * Defines the core types of units we expect to be in any given unit set.
+   * Unit sets should have no less than one of each.
+   * Unit sets should order models such that the most "sensible" type (generally cheapest, land) is the first type of a given role.
+   */
+  public enum UnitRoleEnum
   {
     INFANTRY, MECH,
-    RECON, TANK, MD_TANK, NEOTANK, MEGATANK,
-    APC, ARTILLERY, ROCKETS, PIPERUNNER, ANTI_AIR, MOBILESAM,
-    FIGHTER, BOMBER, STEALTH, STEALTH_HIDE, B_COPTER, T_COPTER, BBOMB,
-    CARRIER, BBOAT, BATTLESHIP, CRUISER, LANDER, SUB, SUB_SUB
+    RECON, ASSAULT,
+    SIEGE, ANTI_AIR,
+    AIR_ASSAULT, AIR_SUPERIORITY,
+    TRANSPORT,
   };
 
+  private static ArrayList<String> unitRoleIDs = null;
+  public static ArrayList<String> getUnitRoleIDs()
+  {
+    if( null == unitRoleIDs )
+    {
+      unitRoleIDs = new ArrayList<String>();
+      for( UnitRoleEnum role : UnitRoleEnum.values() )
+        unitRoleIDs.add(standardizeID(role.toString()));
+    }
+    return unitRoleIDs;
+  }
+
+  public static String standardizeID(String input)
+  {
+    return input.toLowerCase().replaceAll(" ", "_").replaceAll("-", "_");
+  }
+
   // Subs are ships unless they're submerged.
   public enum ChassisEnum
   {
     TROOP, TANK, AIR_LOW, AIR_HIGH, SHIP, SUBMERGED
   };
 
   public String name;
-  public UnitEnum type;
+  public UnitRoleEnum role;
   public ChassisEnum chassis;
   private int moneyCost = 9001;
   public int moneyCostAdjustment = 0;
+  public double customStarValue = 0;
```

Comment:
This is set but never used anywhere. Also, DefendPeace doesn't have any notion of "stars" - you must be thinking of a completely different, entirely unrelated game.

Reply:
I wanted to get the variable in now, since it'd be a massive pain to get it in later.

How about I change it to `customPowerValue`?

Reply:
This is the amount of power a Commander gains when they lose a unit of that type, right? I'm not sure that "custom" is the best word, since every unit gets their own value anyway. Maybe just `abilityPowerValue`?